deployments/charts/kuma/templates/cp-webhooks-and-secrets.yaml

```diff
@@ -261,3 +261,26 @@ webhooks:
         resources:
           - secrets
     sideEffects: None
+  - name: pod.validator.kuma-admission.kuma.io
+    admissionReviewVersions: ["v1"]
+    namespaceSelector:
+      matchLabels:
+        kuma.io/system-namespace: "true"
```

Comment:
I don't think this hook selector is correct. Here you only validate pods that are in Kuma system namespace, meaning - Kuma control plane and Zone Ingress / Zone Egress.

I think we want to validate Pods that will have Kuma DP sidecar injected, right? With this in mind. I think we could just execute validation logic in Kuma injector webhook.

Reply:
Thanks for reviewing. Your point is valid. I'll push a quick change.

Reply:
@jakubdyszkiewicz So I went through `pkg/plugins/runtime/k8s/webhooks/injector/injector.go` and observed that there is an input pod spec and an outputted pod spec (after performing the injection). During injection, a few annotations are added automatically.

So I'm thinking of adding the validation logic which is executed for the input pod spec i.e. inside the [`needInject`](https://github.com/kumahq/kuma/blob/6461cb9511483a4aec566b6b128b76ff946ebe5b/pkg/plugins/runtime/k8s/webhooks/injector/injector.go#L126) function.

Let me know if that sounds correct and I'll push the change.

Reply:
> I think we want to validate Pods that will have Kuma DP sidecar injected, right?

For this, please go through #2331. We might need validation for resources other than pods.

Reply:
I think I would add it even before `needInject` call.

I'm thinking about other objects that we place Kuma annotations and so far I can spot:
* Namespace (injection and mesh annotation)
* Service (protocol annotation)

We can do it in separate PR since your change to validate Pods adds a lot of value already!